core/common/src/main/java/alluxio/wire/TieredIdentity.java

```diff
@@ -194,9 +203,37 @@ public boolean equals(Object o) {
       if (!(o instanceof LocalityTier)) {
         return false;
       }
-      LocalityTier that = (LocalityTier) o;
-      return mTierName.equals(that.mTierName)
-          && Objects.equal(mValue, that.mValue);
+      LocalityTier otherTier = (LocalityTier) o;
+      String otherTierName = otherTier.getTierName();
+      if (!mTierName.equals(otherTierName)) {
+        return false;
+      }
+      String otherTierValue = otherTier.getValue();
+      if (mValue == null && otherTierValue == null) {
+        return true;
+      }
+      if (mValue != null && mValue.equals(otherTierValue)) {
+        return true;
+      }
+      // For node tiers, attempt to resolve hostnames to IP addresses, this avoids common
+      // misconfiguration errors where a worker is using one hostname and the client is using
+      // another.
+      if (Constants.LOCALITY_NODE.equals(mTierName)) {
+        try {
+          String tierIpAddress = NetworkAddressUtils.resolveIpAddress(mValue);
+          String otherTierIpAddress = NetworkAddressUtils.resolveIpAddress(otherTierValue);
+          if (tierIpAddress != null && tierIpAddress.equals(otherTierIpAddress)) {
+            return true;
+          }
+        } catch (IllegalArgumentException e) {
```

Comment:
`IllegalArgumentException` and `NullPointerException` should never happen, but if they do we want to know about it as soon as possible. I suggest letting these exceptions propagate instead of catching and ignoring them

Reply:
Done